edu_edfi_airflow/providers/edfi/transfers/edfi_to_s3.py

```diff
@@ -75,68 +91,99 @@ def execute(self, context) -> str:
         :param context:
         :return:
         """
-        if airflow_util.is_full_refresh(context) and self.api_get_deletes:
-            raise AirflowSkipException(
-                "Skipping delete pull for full_refresh run."
-            )
-
         # If doing a resource-specific run, confirm resource is in the list.
-        config_endpoints = airflow_util.get_config_endpoints(context)  # Returns `[]` if none explicitly specified.
-        if config_endpoints and camel_to_snake(self.resource) not in config_endpoints:
-            raise AirflowSkipException(
-                "Endpoint not specified in DAG config `endpoints`."
-            )
+        config_endpoints = airflow_util.get_config_endpoints(context)
+        if config_endpoints and self.resource not in config_endpoints:
+            raise AirflowSkipException("Endpoint not specified in DAG config endpoints.")
+        
+        # Confirm resource is in XCom-list if passed (used for dynamic XComs retrieved from get-change-version operator).
+        if self.enabled_endpoints and self.resource not in self.enabled_endpoints:
+            raise AirflowSkipException("Endpoint not specified in run endpoints.")
+        # Optionally set destination key by concatting separate args for dir and filename
+        if not self.s3_destination_key:
+            if not (self.s3_destination_dir and self.s3_destination_filename):
+                raise ValueError(
+                    f"Argument `s3_destination_key` has not been specified, and `s3_destination_dir` or `s3_destination_filename` is missing."
+                )
+            self.s3_destination_key = os.path.join(self.s3_destination_dir, self.s3_destination_filename)
 
-        # Force min-change-version if no XCom was found.
-        if self.max_change_version and self.min_change_version is None:
-            self.min_change_version = 0
+        # Check the validity of min and max change-versions.
+        self.check_change_version_window_validity(self.min_change_version, self.max_change_version)
```

Comment:
Move logic from `execute()` into helpers to allow code-sharing in the bulk version of the operator.